npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -252,10 +252,9 @@ def source_for(name, requirement, manifest_name)
 
         return unless resolved_url
         return unless resolved_url.start_with?("http")
-        return if CENTRAL_REGISTRIES.any? { |u| resolved_url.start_with?(u) }
         return if resolved_url.match?(/(?<!pkg\.)github/)
```

Comment:
I wonder why we return here? Seems ok to keep but maybe one to clean up at some point

Reply:
Yes, I was also wondering about that. I can dig around the history for a bit and maybe find out why it's there

Reply:
So, this seems to be there to prevent a subdependency that's a git dependency from getting a `github.com` registry source:

https://github.com/dependabot/dependabot-core/commit/96980b412ba77290f0aef0402c66768e9ef36fe2

It explicitly excludes `pkg.github.com`, so I think it's safe (and necessary) to keep. There's test coverage for this so should be 👍

Reply:
Thanks for digging that out 👍